packages/compass-schema/src/components/field/field.jsx

```diff
@@ -244,7 +244,7 @@ class Field extends Component {
             >
               {get(this.getNestedDocType(), 'fields', []).map((field) => (
                 <div className={nestedFieldStyles} key={field.name}>
-                  <Field
+                  <FieldWithPreferences
```

Comment:
Totally feel free to ignore, a nit: as this is a recursively rendered component, and hoc below adds another element to the rendering tree, maybe it would a bit better here to just pass this one prop that you need directly instead if using a wrapped component inside an unwrapped one (also kinda detaches this component implementation from the wrapper code)